dataplane/linux/wireguard_mgr.go

```diff
@@ -60,33 +66,57 @@ func (m *wireguardManager) OnUpdate(protoBufMsg interface{}) {
 		m.wireguardRouteTable.EndpointRemove(msg.Hostname)
 	case *proto.RouteUpdate:
 		log.WithField("msg", msg).Debug("RouteUpdate update")
-		if msg.Type != proto.RouteType_REMOTE_WORKLOAD {
+		switch msg.Type {
+		case proto.RouteType_LOCAL_WORKLOAD:
+			if cidr := ip.MustParseCIDROrIP(msg.Dst); cidr == nil {
+				log.Errorf("error parsing RouteUpdate CIDR: %s", msg.Dst)
+			} else if cidr.Version() == 4 {
+				log.Debugf("Update for IPv4 CIDR: %s", cidr)
+				m.localCIDRs.Add(cidr)
+				m.wireguardRouteTable.LocalWorkloadCIDRAdd(cidr)
+			}
+		case proto.RouteType_REMOTE_WORKLOAD:
+			if cidr := ip.MustParseCIDROrIP(msg.Dst); cidr == nil {
+				log.Errorf("error parsing RouteUpdate CIDR: %s", msg.Dst)
+			} else if cidr.Version() == 4 {
+				log.Debugf("Update for IPv4 CIDR: %s", cidr)
+				m.wireguardRouteTable.EndpointAllowedCIDRAdd(msg.DstNodeName, cidr)
+			}
+		default:
 			log.Debug("RouteUpdate is not a peer workload update, ignoring")
-			return
-		}
-		cidr := ip.MustParseCIDROrIP(msg.Dst)
-		if cidr != nil {
-			m.wireguardRouteTable.EndpointAllowedCIDRAdd(msg.DstNodeName, cidr)
 		}
 	case *proto.RouteRemove:
 		log.WithField("msg", msg).Debug("RouteRemove update")
 		cidr := ip.MustParseCIDROrIP(msg.Dst)
-		if cidr != nil {
+		if cidr == nil {
+			log.Errorf("error parsing RouteRemove CIDR: %s", msg.Dst)
+		} else if cidr.Version() != 4 {
+			log.Debugf("Route removal for non-IPv4 CIDR: %s", cidr)
```

Comment:
Aren't you just ignoring this case on the add?  Probably don't need to do the remove here?  (Also the route resolver is IPv4 only right now so you won't ever see an IPv6 CIDR)

Reply:
ah yup - didn't mean to do that